app/components/common/root.js

```diff
@@ -27,6 +27,8 @@ class Root extends React.Component {
             <Route path='/login' component={LoginPage} />
             <Route path='/checkout' component={Checkout} />
             <Route exact path='/products/:id' component={SingleProduct} />
+            <Route exact path='/orders/viewall' component={AllOrders} />
```

Comment:
I'm not 100% sure but with this order, the router might confuse the 'viewall' for an id parameter. 
I would suggest just making /orders the url for AllOrders